app/jobs/start_sync_job.rb

```diff
@@ -22,8 +22,20 @@ def perform(sync_id, force: false)
 
   private
 
-  def imports_to_sync?
-    Import.where(state: ImportMicroMachine::FINISHED).any?
+  def should_sync?
+    finished_imports = Import.where(state: ImportMicroMachine::FINISHED)
+
+    return false unless finished_imports
+
+    imports_to_sync = finished_imports.count
+
+    finished_imports.each do |import|
+      errors_count = import.raw_inputs.where('exception is not null').count
+      next unless import.raw_inputs.count == errors_count
+      imports_to_sync -= 1
+    end
+
+    imports_to_sync.positive?
```

Comment:
If that's less than one, then there are no imports to sync, and it returns false.